extension/src/parser.js

```diff
@@ -110,84 +539,14 @@ const grabVariable = (fn, args) => {
   return localStorage.getItem("pageVariable");
 };
 
-// Getting the targeting information. We're careful to cache any results so as to avoid being
-// blocked by facebook.
-let targetingBlocked = false;
-let targetingCache = new Map();
-const getTargeting = ad => {
-  if (ad.targeting) {
-    if (targetingCache.has(ad.targeting))
-      return Promise.resolve({
-        ...ad,
-        targeting: targetingCache.get(ad.targeting)
-      });
-    const url = ad.targeting;
-    delete ad.targeting;
-    if (targetingBlocked) return ad;
-    return new Promise(resolve => {
-      let req = new XMLHttpRequest();
-      req.onreadystatechange = function() {
-        if (req.readyState === 4) {
-          try {
-            const targeting = cleanAd(
-              JSON.parse(req.response.replace("for (;;);", ""))["jsmods"][
-                "markup"
-              ][0][1]["__html"]
-            );
-            if (!targeting) {
-              return resolve(ad);
-            }
-            targetingCache.set(url, targeting);
-            resolve({
-              ...ad,
-              targeting
-            });
-          } catch (e) {
-            targetingBlocked = true;
-            setTimeout(() => (targetingBlocked = false), 15 * 60 * 100);
-            ad.targeting = null;
-            resolve(ad);
-          }
-        }
-      };
-
-      // This is all built out from a close reading of facebook's code.
-      let built = grabVariable(
-        url => {
-          let parsed = new (window.require("URI"))(url);
-          localStorage.setItem("url", url);
-          let req = new window.AsyncRequest()
-            .setURI(url)
-            .setData(parsed)
-            .setMethod("GET")
-            .setRelativeTo(document.body)
-            .setNectarModuleDataSafe(document.body)
```

Comment:
what is this?

Reply:
err, what are `setNectarModuleDataSafe` and `setRelativeTo` in particular?

Reply:
Those are from Facebook, we call them to get some sort of token that allows us to get the targeting without popping up the huge targeting dialog

Reply:
Oh weird.